assets/js-text-with-image.js

```diff
@@ -1,170 +1,222 @@
 /**
- * Text With Image Spacer
+ * Text With Image Spacer Component
  *
- * Ensures that spacer elements in text-with-image sections
- * inherit the correct color palette from the previous section.
- * Optimized for performance with Utils library functions.
+ * Ensures that spacer of text-with-image section gets
+ * the correct color palette from the previous section.
  *
  * @requires js-utils-core.js
  * @requires js-utils.js
  */
-const handleTextWithImageSpacer = (section) => {
-  if (!section) return null;
 
-  const config = {
-    class: {
-      spacer: 'text-image__spacer',
-      palettes: ['palette-one', 'palette-two', 'palette-three']
-    }
-  }
+const TextImageConfig = {
+  selector: {
+    section: '.text-image',
+    spacer: '.text-image__spacer'
+  },
+  palettes: ['palette-one', 'palette-two', 'palette-three']
+}
 
-  // Cache elements but get previous section on demand
-  const getElements = () => ({
-    section,
-    prevSection: $.getSibling(section, '*', 'prev')
-  })
+const TextImageDOM = {
+  elements: {
+    sections: null
+  },
 
-  // Cache checking functions for better performance
-  const hasPalette = (el, palette) => el.classList.contains(palette);
-  const hasSpacerClass = (el) => el.classList.contains(config.class.spacer);
+  cache: {
+    spacers: new Map(),
+    prevSections: new Map()
+  },
 
-  const applyPaletteToSpacer = (spacer, prevChild) => {
-    let foundPrevPalette = null,
-        currentSpacerPalette = null;
+  init() {
+    this.elements.sections = document.querySelectorAll(TextImageConfig.selector.section);
+    return this.elements.sections && this.elements.sections.length > 0;
```

Comment:
checked and `document.querySelectorAll` returns an empty array if none is found. So we can return `this.elements.sections.length > 0` directly without the safety check